### PR TITLE
Get CI running on macOS 11 and 12

### PR DESCRIPTION
**Pull request by dfed**
macOS 10.15 [is no longer supported by Github Actions](https://github.com/actions/runner-images/blob/main/images/macos/macos-10.15-Readme.md). So instead let's run CI on macOS 11 and macOS 12.

### CODE REVIEW

File: Scripts/build.swift [hunk 1]
```diff
@@ -266,7 +224,7 @@ guard let task = Task(rawValue: rawTask) else {
 }
 
 if task.shouldGenerateXcodeProject {
-    try execute(commandPath: "/usr/bin/swift", arguments: ["package", "generate-xcodeproj", "--output=generated/"])
+    try execute(commandPath: "/usr/bin/xcrun", arguments: ["/usr/bin/swift", "package", "generate-xcodeproj", "--output=generated/"])
```
Comment (dfed) [inline Scripts/build.swift:227]: This change is orthogonal, but using `xcrun` means we'll use the `swift` version for the selected Xcode rather than the `swift` version installed on the machine globally.

File: Scripts/build.swift [hunk 2]
```diff
@@ -21,126 +21,91 @@ enum TaskError: Error {
 }
 
 enum Platform: String, CustomStringConvertible {
-    case iOS_10
-    case iOS_11
-    case iOS_12
```
Comment (dfed) [inline Scripts/build.swift:26]: I deleted platforms we aren't testing in CI to clean things up

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -24,49 +23,71 @@ jobs:
       - name: Bundle Install
         run: bundle install
       - name: Select Xcode Version
-        run: sudo xcode-select --switch /Applications/Xcode_12.4.app/Contents/Developer
+        run: xcversion select 12.4
```
Comment (dfed) [inline .github/workflows/ci.yml:26]: This change is effectively just syntactic sugar, but it's easier to maintain.

**Review by fdiaz — APPROVED** [review]

**Review by NickEntin — APPROVED** [review]

**Comment by dfed** [comment]
I've updated our repo settings to make these new CI checks all required. Merging!
